### PR TITLE
Deprecated getGibbonMailer and call GibbonMailer wrapper to remove duplicate code.

### PR DESCRIPTION
**Pull request by crayner**
Deprecated getGibbonMailer and call GibbonMailer wrapper to remove duplicate code.
PHPMailer wrapper updated for use with GMAIL, which requires SMTPSecure = 'ssl'
Refactored Messenger Post Process to use GibbonMailer.

**Bug Fix | New Feature**

## Description
Changes are made to remove duplicate code for use of PHPMailer whilst maintaining backwards compatibility.  I added a @deprecated tag for development, and also added a E_USER_DEPRECATED throw to be logged when this old code is used.  Check php_error log for results.

> [30-Nov-2018 11:15:55 Australia/Sydney] PHP Deprecated:  getGibbonMailer method is deprecated and replaced by Gibbon\Comms\GibbonMailer class.   in F:\websites\crayner\Gibbon\functions.php on line 4648
> [30-Nov-2018 11:15:55 Australia/Sydney] PHP Stack trace:
> [30-Nov-2018 11:15:55 Australia/Sydney] PHP   1. {main}() F:\websites\crayner\Gibbon\modules\Messenger\messenger_postProcess.php:0
> [30-Nov-2018 11:15:55 Australia/Sydney] PHP   2. getGibbonMailer() F:\websites\crayner\Gibbon\modules\Messenger\messenger_postProcess.php:1940
> [30-Nov-2018 11:15:55 Australia/Sydney] PHP   3. trigger_error() F:\websites\crayner\Gibbon\functions.php:4648

GibbonMailer looks for gmail setting and sets SMTPSecure = 'ssl' 
(Could look for port 465 to do this, but this would NOT maintain backwards compatibility.)

messenger_postProcess modified as part of the testing.

## Motivation and Context
The settup did not work for SMTP servers requiring SSL

> [30-Nov-2018 11:11:57 Australia/Sydney] Connection: opened
> [30-Nov-2018 11:12:07 Australia/Sydney] SERVER -> CLIENT: 
> [30-Nov-2018 11:12:07 Australia/Sydney] SMTP NOTICE: EOF caught while checking if connected
> [30-Nov-2018 11:12:07 Australia/Sydney] Connection: closed
> [30-Nov-2018 11:12:07 Australia/Sydney] SMTP Error: Could not authenticate.
> [30-Nov-2018 11:12:07 Australia/Sydney] SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting

## How Has This Been Tested?
Locally  I tested the Messenger Post Process both pre and post changes to ensure that the old call still worked, and that the new call also worked.

Travis

## Screenshots (if appropriate):
No changes to any templates.

### CODE REVIEW

**Comment by crayner** [comment]
Thanks for that..  Just pushing development standards, so other developers
have less to learn if they join the project, hence the deprecation error.
------------------------------------------
I greet you with the great words,
Grace and Peace

Craig Rayner
craig@craigrayner.com
Mobile: +61 456 560 018

**Comment by SKuipers** [comment]
That's a fair consideration 👍  It's tricky, as not all of our users are developers, so if there's a chance that something can cause a breaking change in production we would air on the side of caution.

**Comment by crayner** [comment]
as would I
------------------------------------------
I greet you with the great words,
Grace and Peace

Craig Rayner
craig@craigrayner.com
Mobile: +61 456 560 018

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,5 @@ nbproject/
 /vendor/**/unitTests
 /vendor/**/Examples
 /vendor/**/examples
+
+.idea
```
[the remark this answers is not in the document]
Comment (crayner) [reply .gitignore]: Done

File: functions.php [hunk 2]
```diff
@@ -4641,35 +4642,18 @@ function countLikesByRecipient($connection2, $gibbonPersonIDRecipient, $mode = '
  * @version 1st September 2016
  * @since   1st September 2016
  * @author  Sandra Kuipers
+ * @deprecated 30th Nov 2018
+ * This method has been replaced by the GibbonMailer class, and remains here ONLY to handle legacy calls.
+ * The Deprecation error will be logged, and if asked for in php.ini stop execution.
+ * Productions sites should NOT have display_errors set to true, and therefore not a problem.
```
[the remark this answers is not in the document]
Comment (crayner) [reply functions.php]: So the best of both worlds.  It should never display an error now, but if logging is turned on will log.  An alternative is to check the installType from settings, but even in development it should not stop execution, only warn those who are looking..

**Comment by crayner** [comment]
Not quite what you asked for, but a compromise that should work.

**Comment by SKuipers** [comment]
To be fair, the `ini_get`/`ini_set` is a bit of a clunky compromise ... but at least it's in code that will get removed at some point. I'm going to merge this and test it out.

Looks like another solution that frameworks use is the @ silencer on trigger_error, so that developers can opt-in to the deprecation messages rather than have to manually opt-out by editing their error reporting.

**Comment by crayner** [comment]
The other option is to check the install type from the settings
------------------------------------------
I greet you with the great words,
Grace and Peace

Craig Rayner
craig@craigrayner.com
Mobile: +61 456 560 018

**Pull request merged**